Tools/YamlCreate.ps1

```diff
@@ -160,38 +161,68 @@ Function Read-PreviousWinGet-Manifest {
 }
 
 Function Read-WinGet-InstallerValues {
-    Clear-Variable -Name 'Architecture' -Force -ErrorAction SilentlyContinue
-    Clear-Variable -Name 'InstallerType' -Force -ErrorAction SilentlyContinue
-    Clear-Variable -Name 'InstallerUrl' -Force -ErrorAction SilentlyContinue
-    Clear-Variable -Name 'InstallerSha256' -Force -ErrorAction SilentlyContinue
-    Clear-Variable -Name 'Custom' -Force -ErrorAction SilentlyContinue
-    Clear-Variable -Name 'Silent' -Force -ErrorAction SilentlyContinue
-    Clear-Variable -Name 'SilentWithProgress' -Force -ErrorAction SilentlyContinue
-    Clear-Variable -Name 'ProductCode' -Force -ErrorAction SilentlyContinue
-    Clear-Variable -Name 'Scope' -Force -ErrorAction SilentlyContinue
-    Clear-Variable -Name 'InstallerLocale' -Force -ErrorAction SilentlyContinue
-    Clear-Variable -Name 'UpgradeBehavior' -Force -ErrorAction SilentlyContinue
-    Clear-Variable -Name 'AnotherInstaller' -Force -ErrorAction SilentlyContinue
+    $InstallerValues = @(
+        "Architecture"
+        "InstallerType"
+        "InstallerUrl"
+        "InstallerSha256"
+        "Custom"
+        "Silent"
+        "SilentWithProgress"
+        "ProductCode"
+        "Scope"
+        "InstallerLocale"
+        "UpgradeBehavior"
+        "AnotherInstaller"
+    )
+    Foreach ($InstallerValue in $InstallerValues) {Clear-Variable -Name $InstallerValue -Force -ErrorAction SilentlyContinue}
 
     while ([string]::IsNullOrWhiteSpace($InstallerUrl)) {
         Write-Host
         Write-Host -ForegroundColor 'Green' -Object '[Required] Enter the download url to the installer.'
         $InstallerUrl = Read-Host -Prompt 'Url' | TrimString
     }
+
+    while ([string]::IsNullOrWhiteSpace($SaveOption)) {
+        Write-Host
+        Write-Host -ForegroundColor 'Yellow' -Object "Do you want to save the files to the Temp folder?"
+        $SaveOption = Read-Host -Prompt 'y or n' | TrimString
+    }
```

Comment:
This functionality feels like a good candidate for a function, something like:

$SaveOption = PromptForValue('Yellow', "Do you want to save the files to the Temp folder? (y/n)")

Then you could use it in various places

Reply:
Well it is a one time select under the `Read-WinGet-InstallerValues` Function, so there is no reason for it to be used anywhere else.
If the script gets a overhaul it will be changed.

Reply:
It could be used on line 181 too :) But that's fine, don't sweat it